### DISCUSSION TITLE
ELX does not work with DependencyFactory

### DISCUSSION THREAD

**Discussion by rengit62** [OP-TED/eForms-SDK#663] [Q&A]
Ho Lab
I tested successfully junit test of ElX in project efx-toolkit-java but using DependencyFactory ( and complete fields.json ) instead of 
DependencyFactoryMock in a real  environment in Wildfly , first it is very very slow to get ScriptGenerator and SymbolResolver  : 

![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/e0c7df3f-2cdd-4eb5-98ba-ecaadc365fcc)

and get error on 
 SdkField sdkField = SdkEntityFactory.getSdkField(this.sdkVersion, field);

I copied the class DependencyFactory in efx-toolkit-java in a viewer package 

![image](https://github.com/OP-TED/eForms-SDK/assets/52211037/c270d19e-d1e4-4415-bd75-25681d481636)

then mvn clean install and configured application  pom.xml as : 

	<dependency>
			<groupId>eu.europa.ted.eforms</groupId>
			<artifactId>efx-toolkit-java</artifactId>
			<version>2.0.0-SNAPSHOT</version>
		</dependency>
		<dependency>
			<groupId>eu.europa.ted.eforms</groupId>
			<artifactId>eforms-core-java</artifactId>
			<version>1.0.6-SNAPSHOT</version>
			<exclusions>
				<exclusion>
					<groupId>org.apache.httpcomponents</groupId>
					<artifactId>httpclient</artifactId>
				</exclusion>
				<exclusion>
					<groupId>org.apache.maven</groupId>
					<artifactId>maven-resolver-provider</artifactId>
				</exclusion>
				<exclusion>
					<groupId>org.codehaus.plexus</groupId>
					<artifactId>plexus-utils</artifactId>
				</exclusion>
				<exclusion>
					<groupId>org.codehaus.plexus</groupId>
					<artifactId>plexus-sec-dispatcher</artifactId>
				</exclusion>
				<exclusion>
					<groupId>jakarta.xml.bind</groupId>
					<artifactId>jakarta.xml.bind-api</artifactId>
				</exclusion>
				<exclusion>
					<groupId>org.slf4j</groupId>
					<artifactId>slf4j-api</artifactId>
				</exclusion>
			</exclusions>
		</dependency>
Is there any error in deploy or others ?
Thanks and Regards
Renato

**Comment by rouschr** [comment]
Hello, 

I am not sure how I can help you with a screenshot of your IDE and just one line.
You need to provide more details.
Just by guessing I would say your IDE cannot find one of the classes, so check your pom.xml.
Maybe try using any static method of EFX in a unit test via your IDE to see how well the dependency was integrated.
